File: src/Application.Operation/Features/Employee/Exams/DTOs/ExamConfigurationDto.cs
namespace Application.Operation.Features.Employee.Exams.DTOs;

public sealed class ExamConfigurationDto
{
    public Guid JobId { get; set; }
    public string TitleAr { get; set; } = "";
    public string? TitleEn { get; set; }
    public bool AllowPreviousQuestion { get; set; }
    public Guid InterruptionPolicyId { get; set; }
    public string? Notes { get; set; }
    public int TotalQuestions { get; set; }
    public string? DecisionNotes { get; set; }
    public List<ExamPartDto> Parts { get; set; } = [];
}

File: src/Tawtheef.Websites/operations-web/src/app/pages/user/employee/exams-management/models/exam-configuration.dto.ts
import { ExamPartDto } from "./exam-part.dto";

export interface ExamConfigurationDto {
  jobId: string;
  titleAr: string;
  titleEn: string | null;
  allowPreviousQuestion: boolean;
  interruptionPolicyId: string;
  notes: string | null;
  totalQuestions: number;
  decisionNotes: string | null;
  decisionByName?: string | null;
  decisionAt?: string | null;
  statusBackendName: string | null;
  parts: ExamPartDto[];
}
